README.md

```diff
@@ -112,3 +112,4 @@
 [#53789]: https://github.com/rust-lang/rust/issues/53789
 [18]: https://play.rust-lang.org/?gist=c990af931ab56d93207efcade0ffd01b&version=nightly&mode=debug&edition=2015
 [19]: https://play.rust-lang.org/?gist=1f1d03d4f8280b8843594cba668c9db4&version=nightly&mode=debug&edition=2015
+[20]: https://play.rust-lang.org/?gist=c6a5e5dcf99e04855350770b62528486&version=nightly&mode=debug&edition=2015
```

Comment:
see [discussion on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/122657-wg-nll/subject/crater-run-and-analysis/near/134169102). The TL;DR:

- Diagnostics have an immediate problem of printing `'_` but referencing `'_2`
- Plus the diagnostic is ungreat: it should talk about how `'1` is **local to the closure** but escapes in the return type
- The error itself seems unfortunate but basically correct